projects/packages/jetpack-mu-wpcom/webpack.config.modules.js

```diff
@@ -0,0 +1,61 @@
+/**
+ * Webpack configuration for building JavaScript (ES) modules.
+ */
+const path = require( 'path' );
+const jetpackWebpackConfig = require( '@automattic/jetpack-webpack-config/webpack' );
+const DependencyExtractionWebpackPlugin = require( '@wordpress/dependency-extraction-webpack-plugin' );
+
+const { mode, devtool, output, optimization, resolve } = jetpackWebpackConfig;
+
+/** @type {import('webpack').Configuration} */
+module.exports = {
+	mode,
+	devtool,
+	optimization,
+	resolve,
+	node: false,
+	name: 'jetpack-mu-wpcom/modules',
+	entry: {
+		'code-editor': './src/features/code-editor/code-editor/code-editor.tsx',
+		codemirror: './src/features/code-editor/codemirror/codemirror.ts',
+		'site-additional-css': './src/features/code-editor/site-additional-css/site-additional-css.ts',
+	},
+	output: {
+		...output,
+		filename: '[name]/[name].js',
+		path: path.resolve( __dirname, 'src/build-module' ),
+		module: true,
+		chunkFormat: 'module',
+		asyncChunks: false,
+		environment: { module: true },
+		library: { type: 'module' },
+	},
+	experiments: {
+		outputModule: true,
+	},
+	plugins: [
+		...jetpackWebpackConfig.StandardPlugins( {
+			DependencyExtractionPlugin: false,
+			I18nLoaderPlugin: false,
+			I18nCheckPlugin: false,
+		} ),
+		new DependencyExtractionWebpackPlugin( {
+			requestToExternalModule( request ) {
+				if ( request === '@a8cCodeEditor/codemirror-bundle' ) {
+					return true;
+				}
+			},
+			combineAssets: true,
+		} ),
```

Comment:
Is there a reason not to do like this?
```suggestion
		...jetpackWebpackConfig.StandardPlugins( {
			DependencyExtractionPlugin: {
				requestToExternalModule( request ) {
					if ( request === '@a8cCodeEditor/codemirror-bundle' ) {
						return true;
					}
				},
				combineAssets: true,
			},
			I18nLoaderPlugin: false,
			I18nCheckPlugin: false,
		} ),
```
That would avoid having to have a separate dependency on `@wordpress/dependency-extraction-webpack-plugin`.

Even more of an improvement might be to add a `requestMapModules` option that works like the [`requestMap` option](https://github.com/Automattic/jetpack/tree/trunk/projects/js-packages/webpack-config#dependencyextractionplugin-options-) we have in our wrapper.

Reply:
I took inspiration from the forms webpack config:

https://github.com/Automattic/jetpack/blob/c4dc159f4216c3971db5998ddfb4e2ff4f6d7253/projects/packages/forms/tools/webpack.config.modules.js#L108-L113

I'll see about using the form you mentioned without an additional dependency.

Reply:
[I asked them the same question too](https://github.com/Automattic/jetpack/pull/42011#discussion_r2008290776) (but without a code suggestion). They didn't reply. 😀

Reply:
I applied the suggestion and it seems to be working correctly, thanks 👍